### PR TITLE
[FIX][website] Get website-specific view ids if available

### PR DESCRIPTION
**Pull request by yajo**
In a multiwebsite environment, (de)activating some theme views produces undesired results if the same view exists in different ways for different websites.

With this patch, we search before for a website-specific view with the provided key, and fall back to the XMLID-found one if there is no specific view.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
@Tecnativa

### CODE REVIEW

File: addons/website/controllers/main.py [hunk 1]
```diff
@@ -317,9 +317,15 @@ def seo_suggest(self, keywords=None, lang=None):
...
         ids = []
+        View = request.env["ir.ui.view"]
         for xml_id in xml_ids:
             if "." in xml_id:
-                record_id = request.env.ref(xml_id).id
+                # Get website-specific view if possible
+                record_id = View.search([
+                    ("website_id", "=", request.website.id),
+                    ("key", "=", xml_id),
+                    "|", ("active", "=", True), ("active", "=", False)
```
Comment (pedrobaeza) [inline addons/website/controllers/main.py]: Better to pass context key `active_test=False`

Comment (Yajo) [inline addons/website/controllers/main.py]: Current method does not do that AFAICS

Comment (pedrobaeza) [inline addons/website/controllers/main.py]: I don't get you. This is a generic context key for search. You have to do:
```
View = request.env["ir.ui.view"].with_context(active_test=False)
record_id = View.search([...
```

Comment (Yajo) [inline addons/website/controllers/main.py]: Oh I get you now, OK. I did not know of that feature...

**Comment by pedrobaeza** [comment]
@Gorash maybe you can check this?

**Comment by JKE-be** [comment]
@Yajo 
Not sure about the `.with_context(active_test=False)`
Can you explain the purpose, Seems like that will have side effect in case of just one view deactivated for a specific website.

**Comment by Yajo** [comment]
With that context added, we make the `.search()` get ids for disabled views too. If we don't add it and we have a website-specific view that is disabled, we'd be getting the website-agnostic view id as a result. This is undesired, because we want to be able to enable/disable views specifically for each website separately. If we want the default view to be applied to it, we should drop the specific one instead.

Later in the stack, disabled views get ignored.

**Pull request closed without merging**